### PR TITLE
add an option to enable probability of 429 or timeout response

### PR DESCRIPTION
**Pull request by deads2k**
In production we can run close enough to the line that we occassionally hit the max request limit or a server timeout.  It is very difficult to test these in a "normal" test environment, so our code behaves.... poorly when the happen.

This adds two env vars that can be used (off by default) to give a chance of having particular requests fail in those filters.

@kubernetes/sig-api-machinery-misc @smarterclayton

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go [hunk 1]
```diff
@@ -91,27 +111,34 @@ func WithMaxInFlightLimit(
 
 		if c == nil {
 			handler.ServeHTTP(w, r)
-		} else {
+			return
+		}
+
+		if maxInFlightChaosChance != 0 && rand.Float64() < maxInFlightChaosChance {
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go]: if I am not mistaken, the first condition does not help here. Even with floating points 0.0 will never be smaller than 0.0.

File: staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go [hunk 2]
```diff
@@ -32,6 +36,22 @@ import (
 	"github.com/golang/glog"
 )
 
+var maxInFlightChaosChance = float64(0.0)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go]: a comment with the allowed range would be helpful.

File: staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go (hunk 1 above)
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go]: Forgot my comment. You avoid calling the random num generator.

**Comment by sttts** [comment]
Just a nit about some comment about the allowed range. Otherwise, lgtm.

**Comment by lavalamp** [comment]
Please don't mix test and production code, it will give a false sense of security. I'll let @cheftako continue this argument on my behalf.

**Comment by deads2k** [comment]
>Please don't mix test and production code, it will give a false sense of security. I'll let @cheftako continue this argument on my behalf.

Yeah, more detail is going to be useful here.  This is similar to the prexisting kubelet chaos.

**Comment by deads2k** [comment]
Oh, you probably want to have a different chain.  I think that's likely to be worse than this since it would fall out of date very easily.  We have two today (in kube) and they're hard enough to keep in sync.  These are neatly contained and don't spider out from their single files.

**Comment by dims** [comment]
closing this PR, please reopen if needed

/close

**Pull request closed without merging**